src/irmin-pack/unix/io.ml

```diff
@@ -144,11 +144,11 @@ module Unix = struct
 
   let close t =
     match t.closed with
-    | true -> Error `Double_close
+    | true -> Ok () (* no-op for already closed file *)
     | false -> (
-        t.closed <- true;
         try
           Unix.close t.fd;
+          t.closed <- true;
```

Comment:
@Ngoguey42 here's the :bug:.

If `Unix.close` fails, `t.closed` would still be `true` since it is set beforehand. This is an incorrect state for `t`.

Reply:
Out of curiosity, I just read the [documentation for `close`](https://man7.org/linux/man-pages/man2/close.2.html) more closely and think the current code is correct behavior. If `Unix.close` fails for any reason, it is recommended *not* to retry.

Reply:
Oh great, I did not know that. Can you add a comment inside the code that says that?